qstack/spahm/rho/utils.py

```diff
@@ -70,33 +79,141 @@ def get_xyzlist(xyzlistfile):
 
 
-def load_reps(f_in, from_list=True, single=True, with_labels=False, local=True, reaction=False):
+def load_reps(f_in, from_list=True, single=False, with_labels=False, local=True, summ=False, printlevel=0):
+    '''
+    A function to load representations from txt-list/npy files.
+        Args:
+            - f_in: the name of the input file
+            - fomr_list(bol): if the input file is a txt-file containing a list of paths to the representations
+            - single: if the input file is .npy AND contains a single representation (not an array of representation)
+            - with_label(bol): saves a list of tuple (filename, representation)
+            - local(bol): if the representations is local
+            - summ(bol): if local=True then summs the local components
+            - printlevel(int): level of verbosity
+        Return:
+            np.array with shape (N,M) where N number of representations M dimmensionality of the representation
+            OR tuple (N,np.array(N,M)) containing filenames in pos 0
+    '''
     if from_list:
         X_list = get_xyzlist(f_in)
-        Xs = [np.load(f_X, allow_pickle=True) for f_X in X_list]
+        if printlevel > 0:
+            progress = add_progressbar(max_value=len(X_list)*2)
+            i=0
+        Xs = []
+        for f_X in X_list:
+            Xs.append(np.load(f_X, allow_pickle=True))
+            if printlevel > 0:
+                i+=1
+                progress.update(i)
     else:
-        Xs = [np.load(f_in, allow_pickle=True)]
+        Xs = np.load(f_in, allow_pickle=True) if not single else np.array([np.load(f_in, allow_pickle=True)])
+        if printlevel > 0:
+            progress = add_progressbar(max_value=len(Xs))
+            i=0
     reps = []
+    labels = []
     for x in Xs:
-        labels = []
         if local == True:
-            if  type(x[0,0]) == str:
-                reps.append(x[:,1])
-                labels.append(x[:,0])
+            if x.ndim > 1 and (type(x[0,0]) == str or type(x[0,0]) == np.str_):
+                if summ:
+                    reps.append(x[:,1].sum(axis=0))
+                else:
+                    reps.extend(x[:,1])
+                    labels.extend(x[:,0])
             else:
                 reps.extend(x)
         else:
-           if type(x[0]) == str:
-                reps.extend(x[1])
+           if type(x[0]) == str or type(x[0]) == np.str_:
+                reps.append(x[1])
                 labels.extend(x[0])
            else:
-                reps.extend(x)
+                reps.append(x) 
+        if printlevel > 0:
+            i+=1
+            progress.update(i)
+    if printlevel > 0: progress.finish()
     try:
         reps = np.array(reps, dtype=float)
     except:
+        print(len(reps))
+        shapes = [r.shape[0]  for r in reps]
+        shapes = set(shapes)
+        print(shapes)
         raise RuntimeError("Error while loading representations, check the parameters")
```

Comment:
not sure which type of `Error` it should raise ?

Reply:
I guess runtime is fine

Reply:
:+1: